executor/aggregate.go

```diff
@@ -226,11 +227,15 @@ func (d *HashAggIntermData) getPartialResultBatch(sc *stmtctx.StatementContext,
 // Close implements the Executor Close interface.
 func (e *HashAggExec) Close() error {
 	if e.isUnparallelExec {
-		e.memTracker.Consume(-e.childResult.MemoryUsage())
```

Comment:
Why removing this?

Reply:
Becaue at the end of this Close, there is a `e.memTracker.ReplaceBytesUsed(0)`, so there is really no reason to reduce them here.

Reply:
Well, IMO, it looks more resonable to remove `e.memTracker.ReplaceBytesUsed(0)` instead that can help to track memory leak...

Reply:
Why not keep them both?

Reply:
@XuHuaiyu there is no reason why we need to keep them both:

1. if we want the memTracker to have 0 byte consumption in the end, we should just e.memTracker.ReplaceBytesUsed(0)
2. if we want to detect memory leakage, we should use Consume(-usage) everywhere

@leiysky what I am doing is for 1, but I think 2 should also be a valid point, but I doubt 2 is not satisfied right now, so it could be a future ehancement (we should check more executors), and I would like to add a new ehancement issue for this, does it sounds good to you?

Reply:
For my part, if this modification is aimed to enhance the memory track of `agg.memTracker`, we'd better open another issue or PR to track it.
This modification has nothing to do with the issue that this PR wants to fix.